plugins/field-colour/src/blocks/colourRgb.ts

```diff
@@ -0,0 +1,249 @@
+  const functionName = generator.provideFunction_(
+    'colour_rgb',
+    `
+function ${generator.FUNCTION_NAME_PLACEHOLDER_}(r, g, b)
+  r = math.floor(math.min(100, math.max(0, r)) * 2.55 + .5)
+  g = math.floor(math.min(100, math.max(0, g)) * 2.55 + .5)
+  b = math.floor(math.min(100, math.max(0, b)) * 2.55 + .5)
+  return string.format("#%02x%02x%02x", r, g, b)
+end
+`,
+  );
+  const r = generator.valueToCode(block, 'RED', LuaOrder.NONE) || 0;
+  const g = generator.valueToCode(block, 'GREEN', LuaOrder.NONE) || 0;
+  const b = generator.valueToCode(block, 'BLUE', LuaOrder.NONE) || 0;
+  const code = functionName + '(' + r + ', ' + g + ', ' + b + ')';
+  return [code, LuaOrder.HIGH];
+}
+
+/**
+ * PHP block generator function.
+ *
+ * @param block The Block instance to generate code for.
+ * @param generator The PhpGenerator calling the function.
+ * @returns A tuple containing the code string and precedence.
+ */
+export function phpGenerator(
+  block: Block,
+  generator: PhpGenerator,
+): [string, PhpOrder] {
+  // Compose a colour from RGB components expressed as percentages.
+  const red = generator.valueToCode(block, 'RED', PhpOrder.NONE) || 0;
+  const green = generator.valueToCode(block, 'GREEN', PhpOrder.NONE) || 0;
+  const blue = generator.valueToCode(block, 'BLUE', PhpOrder.NONE) || 0;
+  const functionName = generator.provideFunction_(
+    'colour_rgb',
+    `
+function ${generator.FUNCTION_NAME_PLACEHOLDER_}($r, $g, $b) {
+  $r = round(max(min($r, 100), 0) * 2.55);
+  $g = round(max(min($g, 100), 0) * 2.55);
+  $b = round(max(min($b, 100), 0) * 2.55);
+  $hex = '#';
+  $hex .= str_pad(dechex($r), 2, '0', STR_PAD_LEFT);
+  $hex .= str_pad(dechex($g), 2, '0', STR_PAD_LEFT);
+  $hex .= str_pad(dechex($b), 2, '0', STR_PAD_LEFT);
+  return $hex;
+}
+`,
+  );
+  const code = functionName + '(' + red + ', ' + green + ', ' + blue + ')';
+  return [code, PhpOrder.FUNCTION_CALL];
+}
+
+/**
+ * Python block generator function.
+ *
+ * @param block The Block instance to generate code for.
+ * @param generator The PythonGenerator calling the function.
+ * @returns A tuple containing the code string and precedence.
+ */
+export function pythonGenerator(
+  block: Block,
+  generator: PythonGenerator,
+): [string, PythonOrder] {
+  // Compose a colour from RGB components expressed as percentages.
+  const functionName = generator.provideFunction_(
+    'colour_rgb',
+    `
+def ${generator.FUNCTION_NAME_PLACEHOLDER_}(r, g, b):
+  r = round(min(100, max(0, r)) * 2.55)
+  g = round(min(100, max(0, g)) * 2.55)
+  b = round(min(100, max(0, b)) * 2.55)
+  return '#%02x%02x%02x' % (r, g, b)
+`,
+  );
+  const r = generator.valueToCode(block, 'RED', PythonOrder.NONE) || 0;
+  const g = generator.valueToCode(block, 'GREEN', PythonOrder.NONE) || 0;
+  const b = generator.valueToCode(block, 'BLUE', PythonOrder.NONE) || 0;
+  const code = functionName + '(' + r + ', ' + g + ', ' + b + ')';
```

Comment:
```suggestion
  const code = `${functionName}(${r}, ${g}, ${b})`;
```
(but see above re: naming.)

Reply:
Tracked in #2198 for follow-up.